comp/process/agent/agentimpl/agent.go

```diff
@@ -81,7 +94,16 @@ func newProcessAgent(p processAgentParams) agent.Component {
 		Log:     p.Log,
 	}
 
-	return processAgentComponent
+	if flavor.GetFlavor() != flavor.ProcessAgent {
```

Comment:
Code in `comp/process/agent` is specific to the process Agent, isn't it? When `flavor.GetFlavor()` is different from `flavor.ProcessAgent`?

Reply:
Currently, we are in the process of migrating the process checks the core agent thus the component can actually be used in the core agent.